internal/provider/resource_id.go

```diff
@@ -4,17 +4,19 @@ import (
 	"context"
 	"crypto/rand"
 	"encoding/base64"
-	"encoding/hex"
+	hex2 "encoding/hex"
```

Comment:
In general with Go, it is better to avoid renaming imports, especially those from standard library. Instead I'd recommend renaming the variable that's trying to use `hex` instead. 👍 

Reference: https://github.com/golang/go/wiki/CodeReviewComments#imports

Reply:
Ooops. My bad. Fixed.